.ado/jobs/react-native-test-app-integration.yml

```diff
@@ -9,83 +9,49 @@ jobs:
-        displayName: Modify react-native-test-app dependencies
+
+      - template: /.ado/templates/verdaccio-publish.yml@self
+
+      - script: |
+          git clone --filter=blob:none --progress https://github.com/microsoft/react-native-test-app.git
+        displayName: Clone react-native-test-app
+
+      - script: |
+          node .ado/scripts/export-versions.mjs
+        displayName: Determine react-native version
+
+      - script: |
+          npm run set-react-version $(react_native_version) -- --overrides '{ "react-native-macos": "1000.0.0" }'
+        displayName: Configure react-native-test-app dependencies
         workingDirectory: react-native-test-app
-      - bash: |
-          set -eo pipefail
-          cat package.json |
-            jq '.devDependencies["@react-native-community/cli"] = $(rncli_version)' |
-            jq '.devDependencies["@react-native-community/cli-platform-android"] = $(rncli_android_version)' |
-            jq '.devDependencies["@react-native-community/cli-platform-ios"] = $(rncli_ios_version)' |
-            jq '.devDependencies["react"] = $(react_version)' |
-            jq '.devDependencies["react-native"] = "^0.71"' |
-            jq '.devDependencies["react-native-macos"] = "../../react-native-macos-$(package_version).tgz"' |
-            jq 'del(.devDependencies["react-native-windows"])' > .package.json
-          mv .package.json package.json
-          cat package.json | jq .devDependencies
-        displayName: Modify example app dependencies
-        workingDirectory: react-native-test-app/example
-      - template: /.ado/templates/verdaccio-init.yml@self
-      - bash: |
-          npx beachball publish --branch origin/$(System.PullRequest.TargetBranch) --no-push --registry http://localhost:4873 --yes --access public
-        displayName: Publish beachball packages to verdaccio
-      - bash: |
-          cat .yarnrc.yml | sed 's_^npmRegistryServer: ".*"$_npmRegistryServer: "http://localhost:4873"_' > .yarnrc.yml.copy
-          rm .yarnrc.yml
-          mv .yarnrc.yml.copy .yarnrc.yml
-          echo -e '\nunsafeHttpWhitelist: ["localhost"]' >> .yarnrc.yml
-        displayName: Point react-native-test-app registry to verdaccio server
-        workingDirectory: react-native-test-app
-      - bash: |
+
+      - script: |
+          $(Build.Repository.LocalPath)/.ado/scripts/verdaccio.sh configure
           yarn --no-immutable
         displayName: Install npm dependencies
         workingDirectory: react-native-test-app
-      - bash: |
+
+      - script: |
           yarn build:macos || yarn build:macos
         displayName: Bundle JavaScript
         workingDirectory: react-native-test-app/example
-      - bash: |
+
+      - script: |
           rm macos/Podfile.lock
           pod install --project-directory=macos
         displayName: Install Pods
         workingDirectory: react-native-test-app/example
-      - bash: |
-          set -eo pipefail
```

Comment:
I've been adding `set -eox pipefail` everywhere. Does the `script` task do that automatically or something?

Reply:
I thought it did, but I must've misread. We probably want this instead: https://github.com/microsoft/fluentui/pull/17459/files

Reply:
I added the script to a template that gets included in most pipelines.

Reply:
I think that made every pod install fail. We can probably undo that.

Reply:
Yeah, it's oversensitive now. I'm currently looking into tweaking the fix a bit. Worst case, I'll revert it and fix it in a separate PR.

Reply:
Okay, we certainly learned that the hard way. I've put the `set -eo pipefail` lines back in.